File: Mbs.Api/Models/Trading/Instruments/InstrumentReference.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;

namespace Mbs.Api.Models.Trading.Instruments
{
    /// <summary>
    /// Contains information to reference an instrument.
    /// </summary>
    [DataContract]
    public class InstrumentReference : IValidatableObject
    {
        /// <summary>
        /// Gets or sets an exchange representations according to ISO 10383 Market Identifier Code (MIC).
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string Mic { get; set; }

        /// <summary>
        /// Gets or sets an ISIN.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string Isin { get; set; }

        /// <summary>
        /// Gets or sets the optional symbol (ticker) of the security.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string Symbol { get; set; }

        /// <summary>
        /// Gets or sets a name of the instrument.
        /// </summary>
        [DataMember(IsRequired = false, EmitDefaultValue = false)]
        public string Name { get; set; }

        /// <inheritdoc />
        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }
}

File: Mbs.Api.Host.Ng/ClientApp/src/app/shared/mbs/instruments/types/instrument-reference.ts
/** Contains information to reference an instrument. */
export class InstrumentReference {
  /** An exchange representations according to ISO 10383 *Market Identifier Code* (MIC). */
  mic?: string;

  /** An ISIN. */
  isin?: string;

  /** An optional symbol (ticker) of the security. */
  symbol?: string;

  /** A name of the instrument. */
  name?: string;

  constructor(data?: InstrumentReference) {
    if (data) {
      for (const property in data) {
        if (data.hasOwnProperty(property)) {
          (this as any)[property] = (data as any)[property];
        }
      }
    }
  }

  toJSON(data?: any) {
    data = typeof data === 'object' ? data : {};
    /* eslint-disable @typescript-eslint/dot-notation */
    data['Mic'] = this.mic;
    data['Isin'] = this.isin;
    data['Symbol'] = this.symbol;
    data['Name'] = this.name;
    /* eslint-enable @typescript-eslint/dot-notation */
    return data;
  }
}
